addons/website_crm_partner_assign_commissions/models/account_invoice.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, _
+from odoo.tools import float_compare
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    reseller_id = fields.Many2one('res.partner', 'Reseller', help='This field is used to track the reseller in order to generate commisions')
+    purchase_order_line_id = fields.Many2one('purchase.order.line', 'Reseller Purchase Order line')
+    commission_ratio = fields.Float('Commission Ratio', default=1, help="1 means that all the commission goes to the partner")
+
+    def _make_commissions(self):
+        """
+            Adds a line to a purchase order for reseller_id containing the due commissions for the invoice
+            The line is only added the first time we call this method (called on reconcile)
+        """
+        self.ensure_one()
+
+        if not self.reseller_id or self.purchase_order_line_id:
+            return
+
+        total = 0
+        for commission in self.reseller_id.grade_id.commission_ids:
+            lines = self.invoice_line_ids.filtered(lambda line: line.product_id.product_tmpl_id.id in commission.product_ids.ids)
+            total += sum(line.price_subtotal * self.commission_ratio * commission.percentage / 100 for line in lines)
```

Comment:
Are your sure about this formula?
If I recall correctly, the `commission_ratio` is here to override the partner grade percentage.

Shouldn't that be something like this

```python
    for commission in self.reseller_id.grade_id.commission_ids:
        commission_rate = self.commission_ratio or commission.percentage
        lines = self.invoice_line_ids.filtered(lambda line: line.product_id.product_tmpl_id.id in commission.product_ids.ids)
        total += sum(line.price_subtotal * (commission_rate / 100.0) for line in lines)
```

(If you take in account my review on the enterprise PR with the rate being between 0 and 100)